language-plutus-core/test/Evaluation/CkMachine.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Evaluation.CkMachine
+    ( test_evaluateCk
+    ) where
+
+import           PlutusPrelude hiding (hoist, list)
+import           Language.PlutusCore
+import           Language.PlutusCore.Constant
+import           Language.PlutusCore.CkMachine
+import           Language.PlutusCore.TestSupport
+import           Language.PlutusCore.StdLib.Data.Bool
+import           Language.PlutusCore.StdLib.Data.Function
+import           Language.PlutusCore.StdLib.Data.List
+import           Language.PlutusCore.StdLib.Data.Nat
+import           Language.PlutusCore.StdLib.Data.Unit
+
+import           Data.Foldable
+import           Control.Monad.Morph
+import           Hedgehog hiding (Size, Var, annotate)
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import           Test.Tasty
+import           Test.Tasty.Hedgehog
+
+-- | Convert an 'Integer' to a @nat@.
+getBuiltinIntegerToNat :: Integer -> Quote (Term TyName Name ())
```

Comment:
Why not put these in the stdlib?

Reply:
I plan to remove that once PLC `integer`s are indexed by size. Currently there are several such hand-wavy places where I do something not too principled just to get things work. There was also some discussion about adding a new primitive: `sizeOfInteger : integer s -> size s` which should simplify things, so I do not want to touch terms with sizes right now.